tools/apiview/parsers/cpp-api-parser/ApiViewProcessor/AstDumper.hpp

```diff
@@ -24,7 +24,10 @@ class AstDumper {
       std::vector<std::string>::iterator& current);
 
 protected:
-  void UpdateCursor(size_t cursorAdjustment) { m_currentCursor += cursorAdjustment; }
+  void UpdateCursor(size_t cursorAdjustment) {
+      m_currentCursor += cursorAdjustment; 
+      assert(m_currentCursor >= 0);
```

Comment:
This assert will always be true since size_t is an unsigned int.  Instead you'd need to see if cursorAdjustment is > max(size_t)-m_currentCursor.  The more I think about it, the less I believe checking matters :)